Bootnode peer discovery fails

# Generated bootnode id seems to be mismatched with the actually to-be-discovered node id.
## Description
I am trying to build up a private network with docker compose.

For that I use a modified version of this repo's: 

- Dockerfile. (See mine attached. I had to rename to .txt bc it did not allowed me to upload)
- polygon-edge.sh (See mine attached. I had to rename to .txt bc it did not allowed me to upload)
- and docker-compose.yml (See mine attached. Only difference is that it uses (my local Docerfile, see dockerfile diffs in Dockerfile. Also .yml renamed to .txt)
[docker-compose.txt](https://github.com/0xPolygon/polygon-edge/files/10344268/docker-compose.txt)
[Dockerfile.txt](https://github.com/0xPolygon/polygon-edge/files/10344269/Dockerfile.txt)
[polygon-edge.txt](https://github.com/0xPolygon/polygon-edge/files/10344270/polygon-edge.txt)


When I attach to those nodes, all are complaining about this:

```
2023-01-04T12:23:34.835Z [ERROR] polygon.network.discovery: Unable to execute bootnode peer discovery: bootnode=16Uiu2HAkxgphSqizpEDUUWLospTg7urqNzk4Jh7ygYNRovaMyWae err="unable to create new discovery client connection, could not initialize new discovery client - peer [16Uiu2HAkxgphSqizpEDUUWLospTg7urqNzk4Jh7ygYNRovaMyWae] not connected"
2023-01-04T12:24:34.835Z [ERROR] polygon.network.discovery: Unable to execute bootnode peer discovery: bootnode=16Uiu2HAm7LM5kexwkjy1M4R1hKcptCivX5bPcxgtd7mxrQYQrnsr err="unable to create new discovery client connection, could not initialize new discovery client - peer [16Uiu2HAm7LM5kexwkjy1M4R1hKcptCivX5bPcxgtd7mxrQYQrnsr] not connected
```

Tho, when I attach all the nodes and query each of their secrets and node id, there are no nodes with node id:

```
16Uiu2HAm7LM5kexwkjy1M4R1hKcptCivX5bPcxgtd7mxrQYQrnsr
16Uiu2HAkxgphSqizpEDUUWLospTg7urqNzk4Jh7ygYNRovaMyWae
```

My genesis file tho, looks correct (with these 2 bootnodes above).
```
{
    "name": "polygon-edge",
    "genesis": {
        "nonce": "0x0000000000000000",
        "timestamp": "0x0",
        ...
        ...
    "params": {
        "forks": {
            "homestead": 0,
            "byzantium": 0,
            "constantinople": 0,
            "petersburg": 0,
            "istanbul": 0,
            "EIP150": 0,
            "EIP158": 0,
            "EIP155": 0
        },
        "chainID": 100,
        "engine": {
            "ibft": {
                "epochSize": 100000,
                "type": "PoA",
                "validator_type": "bls"
            }
        },
        "blockGasTarget": 0
    },
    "bootnodes": [
        "/dns4/node-1/tcp/1478/p2p/16Uiu2HAkxgphSqizpEDUUWLospTg7urqNzk4Jh7ygYNRovaMyWae",
        "/dns4/node-2/tcp/1478/p2p/16Uiu2HAm7LM5kexwkjy1M4R1hKcptCivX5bPcxgtd7mxrQYQrnsr"
    ]
}
```

Could you please tell me where are these 'uknown' node ids coming from ?

## Your environment
- MacOSX - Docker compose 3.5
- The version of the Polygon Edge:   
  [VERSION INFO]
Release version = v0.6.1

## Steps to reproduce
- Tell us how to reproduce this issue.
1. Have docker and docker compose installed
2. Place these 3 files next to each other and run docker build:
`docker build -t my_polygon_edge_node .`
3. `docker compose up -d`

## Expected behavior
You shall be seeing the same discovery issue. Most probably i'm configuring something wrong I guess, but I do not exactly see where.

Thank you in advance for your help!

[log_node1.txt](https://github.com/0xPolygon/polygon-edge/files/10344242/log_node1.txt)
[log_node2.txt](https://github.com/0xPolygon/polygon-edge/files/10344243/log_node2.txt)
[log_node3.txt](https://github.com/0xPolygon/polygon-edge/files/10344244/log_node3.txt)
[log_node4.txt](https://github.com/0xPolygon/polygon-edge/files/10344245/log_node4.txt)

--- a/docker/local/docker-compose.yml
+++ b/docker/local/docker-compose.yml
@@ -8,11 +8,7 @@ services:
       dockerfile: docker/local/Dockerfile
     command: ["init"]
     volumes:
-      - node-1:/polygon-edge/data-1
-      - node-2:/polygon-edge/data-2
-      - node-3:/polygon-edge/data-3
-      - node-4:/polygon-edge/data-4
-      - genesis:/genesis
+      - data:/data
     networks:
       - polygon-edge-docker
   
@@ -22,7 +18,7 @@ services:
     build:
       context: ../../
       dockerfile: docker/local/Dockerfile
-    command: ["server", "--data-dir", "/data", "--chain", "/genesis/genesis.json", "--grpc-address", "0.0.0.0:9632", "--libp2p", "0.0.0.0:1478", "--jsonrpc", "0.0.0.0:8545", "--prometheus", "0.0.0.0:5001", "--seal"]
+    command: ["server", "--data-dir", "/data/data-1", "--chain", "/data/genesis.json", "--grpc-address", "0.0.0.0:9632", "--libp2p", "0.0.0.0:1478", "--jsonrpc", "0.0.0.0:8545", "--prometheus", "0.0.0.0:5001", "--seal"]
     depends_on:
       init:
         condition: service_completed_successfully
@@ -31,8 +27,7 @@ services:
       - '10002:8545'
       - '10003:5001'
     volumes:
-      - node-1:/data
-      - genesis:/genesis
+      - data:/data
     networks:
       - polygon-edge-docker
     restart: on-failure
@@ -41,7 +36,7 @@ services:
     build:
       context: ../../
       dockerfile: docker/local/Dockerfile
-    command: ["server", "--data-dir", "/data", "--chain", "/genesis/genesis.json", "--grpc-address", "0.0.0.0:9632", "--libp2p", "0.0.0.0:1478", "--jsonrpc", "0.0.0.0:8545", "--prometheus", "0.0.0.0:5001", "--seal"]
+    command: ["server", "--data-dir", "/data/data-2", "--chain", "/data/genesis.json", "--grpc-address", "0.0.0.0:9632", "--libp2p", "0.0.0.0:1478", "--jsonrpc", "0.0.0.0:8545", "--prometheus", "0.0.0.0:5001", "--seal"]
     depends_on:
       init:
         condition: service_completed_successfully
@@ -50,8 +45,7 @@ services:
       - '20002:8545'
       - '20003:5001'
     volumes:
-      - node-2:/data
-      - genesis:/genesis
+      - data:/data
     networks:
       - polygon-edge-docker
     restart: on-failure
@@ -60,7 +54,7 @@ services:
     build:
       context: ../../
       dockerfile: docker/local/Dockerfile
-    command: ["server", "--data-dir", "/data", "--chain", "/genesis/genesis.json", "--grpc-address", "0.0.0.0:9632", "--libp2p", "0.0.0.0:1478", "--jsonrpc", "0.0.0.0:8545", "--prometheus", "0.0.0.0:5001", "--seal"]
+    command: ["server", "--data-dir", "/data/data-3", "--chain", "/data/genesis.json", "--grpc-address", "0.0.0.0:9632", "--libp2p", "0.0.0.0:1478", "--jsonrpc", "0.0.0.0:8545", "--prometheus", "0.0.0.0:5001", "--seal"]
     depends_on:
       init:
         condition: service_completed_successfully
@@ -69,8 +63,7 @@ services:
       - '30002:8545'
       - '30003:5001'
     volumes:
-      - node-3:/data
-      - genesis:/genesis
+      - data:/data
     networks:
       - polygon-edge-docker
     restart: on-failure
@@ -79,7 +72,7 @@ services:
     build:
       context: ../../
       dockerfile: docker/local/Dockerfile
-    command: ["server", "--data-dir", "/data", "--chain", "/genesis/genesis.json", "--grpc-address", "0.0.0.0:9632", "--libp2p", "0.0.0.0:1478", "--jsonrpc", "0.0.0.0:8545", "--prometheus", "0.0.0.0:5001", "--seal"]
+    command: ["server", "--data-dir", "/data/data-4", "--chain", "/data/genesis.json", "--grpc-address", "0.0.0.0:9632", "--libp2p", "0.0.0.0:1478", "--jsonrpc", "0.0.0.0:8545", "--prometheus", "0.0.0.0:5001", "--seal"]
     depends_on:
       init:
         condition: service_completed_successfully
@@ -88,8 +81,7 @@ services:
       - '40002:8545'
       - '40003:5001'
     volumes:
-      - node-4:/data
-      - genesis:/genesis
+      - data:/data
     networks:
       - polygon-edge-docker
     restart: on-failure
@@ -100,8 +92,5 @@ networks:
     name: polygon-edge-docker
 
 volumes:
-  node-1:
-  node-2:
-  node-3:
-  node-4:
+  data:
   genesis:

--- a/docker/local/polygon-edge.sh
+++ b/docker/local/polygon-edge.sh
@@ -2,8 +2,8 @@
 
 set -e
 
-POLYGON_EDGE_BIN=./polygon-edge
-GENESIS_PATH=/genesis/genesis.json
+POLYGON_EDGE_BIN=/polygon-edge/polygon-edge
+GENESIS_PATH=/data/genesis.json
 
 case "$1" in
 
@@ -12,11 +12,11 @@ case "$1" in
           echo "Secrets have already been generated."
       else
           echo "Generating secrets..."
-          secrets=$("$POLYGON_EDGE_BIN" secrets init --num 4 --data-dir data- --json)
+          secrets=$("$POLYGON_EDGE_BIN" secrets init --num 4 --data-dir /data/data- --json)
           echo "Secrets have been successfully generated"
 
           echo "Generating genesis file..."
-          "$POLYGON_EDGE_BIN" genesis \
+          cd /data && "$POLYGON_EDGE_BIN" genesis \
             --dir "$GENESIS_PATH" \
             --consensus ibft \
             --ibft-validators-prefix-path data- \

